src/shared/settingsConfiguration.ts

```diff
@@ -56,4 +56,50 @@ export class DefaultSettingsConfiguration implements SettingsConfiguration {
             return false
         }
     }
+    /**
+     * Sets a prompt message as suppressed.
+     * @param promptName Name of prompt to append to list
+     */
+    public disablePrompt(promptName: string): void {
+        const setting = this.getSuppressPromptSetting(promptName)
+        if (setting === undefined || setting[promptName]) {
+            return
+        }
+        setting[promptName] = true
+        this.writeSetting('suppressPrompts', setting, vscode.ConfigurationTarget.Global)
+    }
+    /**
+     * Verifies if a prompt should be displayed again.
+     * @param promptName Name of the prompt
+     * @returns False when prompt has been suppressed
+     */
+    public shouldDisplayPrompt(promptName: string): boolean {
+        const promptSetting = this.getSuppressPromptSetting(promptName)
+        if (promptSetting !== undefined && promptSetting[promptName]) {
+            return false
+        }
+        return true
+    }
+
+    /**
+     * Gets the 'aws.suppressPrompts' setting. This will reset the setting to default if it does not
+     * recieve an object or the prompts value is not a boolean.
+     * @param promptName
+     * @returns The settings object
+     */
+    public getSuppressPromptSetting(promptName: string): { [prompt: string]: boolean } | undefined {
+        try {
+            const setting = this.readSetting<{ [prompt: string]: boolean }>('suppressPrompts')
+            if (setting !== undefined && (typeof setting !== 'object' || typeof setting[promptName] !== 'boolean')) {
+                getLogger().warn('Setting "suppressPrompts" has an unexpected type. Resetting to default.')
+                // writing this setting to an empty object reverts the setting to its default
+                this.writeSetting('suppressPrompts', {}, vscode.ConfigurationTarget.Global)
+                return undefined
```

Comment:
We can just return `false` here. That simplifies the return type and simplifies caller logic. 

`undefined` could be useful if callers were interested in whether the setting existed at all. But I don't think there are such cases.

Reply:
`False` will not work here. The purpose of this function is to return the setting object, which has all the prompts. Unlike our other settings, we can't just overwrite it.  We need to get the object, update the one prompt, then write that back to settings. We should use the `shouldDisplayPrompt` function when we need to determine if a prompt is suppressed.

Reply:
oops i misread the return type.